testnet/stacks-node/src/nakamoto_node/sign_coordinator.rs

```diff
@@ -250,7 +254,28 @@ impl SignCoordinator {
             ..Default::default()
         };
 
-        let mut coordinator: FireCoordinator<Aggregator> = FireCoordinator::new(coord_config);
+        let total_weight = reward_set.total_signing_weight().map_err(|e| {
+            warn!("Failed to calculate total weight for the reward set: {e:?}");
+            ChainstateError::NoRegisteredSigners(0)
+        })?;
+
+        let threshold = NakamotoBlockHeader::compute_voting_weight_threshold(total_weight)?;
+
+        let signer_public_keys = reward_set_signers
+            .iter()
+            .cloned()
+            .enumerate()
+            .map(|(idx, signer)| {
+                let Ok(slot_id) = u32::try_from(idx) else {
+                    return Err(ChainstateError::InvalidStacksBlock(
+                        "Signer index exceeds u32".into(),
+                    ));
+                };
+                Ok((slot_id, signer))
+            })
+            .collect::<Result<HashMap<_, _>, ChainstateError>>()?;
+
+        let coordinator: FireCoordinator<Aggregator> = FireCoordinator::new(coord_config);
```

Comment:
Can this be moved into the `#[cfg(test)]` block?  Also, can the imports for wsts be moved there as well?

Reply:
Both the `cfg(test)` block and the "non test" block return a `SignCoordinator`, which has a `coordinator` field. So, we'd have to return some other type if we wanted to only include the wsts stuff in the `cfg(test)` block. Ultimately, this whole file relies on both the v0 and v1 stuff (ie `begin_sign_v0` and `begin_sign_v1`) - there's no easy fix other than separate files / structs IMO